conformance/tests/httproute-observed-generation-bump.go

```diff
@@ -0,0 +1,114 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteObservedGenerationBump)
+}
+
+var HTTPRouteObservedGenerationBump = suite.ConformanceTest{
+	ShortName:   "HTTPRouteObservedGenerationBump",
+	Description: "A HTTPRoute in the gateway-conformance-infra namespace should update the observedGeneration in all of it's Status.Conditions after an update to the spec",
+	Manifests:   []string{"tests/httproute-observed-generation-bump.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+
+		routeNN := types.NamespacedName{Name: "observed-generation-bump", Namespace: "gateway-conformance-infra"}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: "gateway-conformance-infra"}
+
+		acceptedCondition := metav1.Condition{
+			Type:   string(v1beta1.RouteConditionAccepted),
+			Status: metav1.ConditionTrue,
+			Reason: "", // any reason
+		}
+
+		t.Run("observedGeneration should increment", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+
+			namespaces := []string{"gateway-conformance-infra"}
+			kubernetes.NamespacesMustBeAccepted(t, s.Client, s.TimeoutConfig, namespaces)
+
+			existing := &v1beta1.HTTPRoute{}
+			err := s.Client.Get(ctx, routeNN, existing)
+			require.NoErrorf(t, err, "error getting HTTPRoute: %v", err)
+
+			// Sanity check
+			if kubernetes.HTTPRouteConditionsHaveLatestObservedGeneration(existing) {
+				t.Fatal("Not all the condition's observedGeneration were updated")
+			}
+
+			existing.Spec.Rules[0].BackendRefs[0].Name = "infra-backend-v2"
+			err = s.Client.Update(ctx, existing)
+			require.NoErrorf(t, err, "error updating the HTTPRoute: %v", err)
+
+			kubernetes.HTTPRouteMustHaveCondition(t, s.Client, s.TimeoutConfig, routeNN, gwNN, acceptedCondition)
+
+			updated := &v1beta1.HTTPRoute{}
+			err = s.Client.Get(ctx, routeNN, updated)
+			require.NoErrorf(t, err, "error getting Gateway: %v", err)
+
+			// Sanity check
+			if kubernetes.HTTPRouteConditionsHaveLatestObservedGeneration(updated) {
+				t.Fatal("Not all the condition's observedGeneration were updated")
+			}
+
+			if existing.Generation == updated.Generation {
+				t.Errorf("Expected generation to change because of spec change - remained at %v", updated.Generation)
+			}
+
+			for _, up := range updated.Status.Parents {
+				existing := parentStatusForRef(existing.Status.Parents, up.ParentRef)
```

Comment:
This check never works when I run it because I always get an empty list of parent statuses coming from `existing`, I think we need to examine exactly what state of the route ends up in the `expected` object since `Update` will update the content that `expected` points to

```
	// Update updates the given obj in the Kubernetes cluster. obj must be a
	// struct pointer so that obj can be updated with the content returned by the Server.
	Update(ctx context.Context, obj Object, opts ...UpdateOption) error
```

can we change this to just assert on the exact set of parent refs we expect? since we know what the route is we also know what the parentrefs should be from the fixture